### DISCUSSION TITLE
Citeseer graph discrepancy

### DISCUSSION THREAD

**Discussion by qubit0** [pyg-team/pytorch_geometric#6914] [Q&A]
Why is there a discrepancy between citeseer graph from [PyG](https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.datasets.Planetoid.html#torch_geometric.datasets.Planetoid) and [DGL](https://docs.dgl.ai/en/1.0.x/generated/dgl.data.CiteseerGraphDataset.html#:~:text=In%20citeseer%20dataset%2C%20there%20are%20some%20isolated%20nodes,added%20as%20zero-vecs%20into%20the%20right%20position.%20Examples)?

It seems like PyG gets rid of self-loop from citeseer graph. The extra 124 edges reported by DGL is the number of self-loop in the citeseer graph.

**Comment by qubit0** [comment]
It seems like PyG did some [clean-up](https://github.com/pyg-team/pytorch_geometric/issues/6119). The reason for getting rid of the self-loop is that for the citation network, the self-loop doesn't make sense as it would mean the paper is citing itself, right?

**Reply by rusty1s** [reply]
Yes, we do some post-processing to clean up the dataset. In the end, it doesn't make a big difference.
